client/clientset/fake/extensions.go

```diff
@@ -31,11 +32,11 @@ func NewFakeExtensionClient(objects ...runtime.Object) *FakeExtensionClient {
 	return &FakeExtensionClient{&fakePtr}
 }
 
-func (m *FakeExtensionClient) DatabaseSnapshots(ns string) client.DatabaseSnapshotInterface {
+func (m *FakeExtensionClient) DatabaseSnapshot(ns string) client.DatabaseSnapshotInterface {
```

Comment:
I vote is for whatever the convention is. `DatabaseSnapshots` (plural) is the right convention. Update the interface and also add checks like `var _ interface = &struct{]`